File: src/main/java/com/github/weewar/mapviewer/model/enums/UnitType.java
package com.github.weewar.mapviewer.model.enums;

public enum UnitType {
    antiAircraft,
    assaultArtillery,
    battleship,
    berserker,
    bomber,
    destroyer,
    dfa,
    heavyArtillery,
    heavyTank,
    heavyTrooper,
    helicopter,
    hovercraft,
    jet,
    lightArtillery,
    raider,
    speedboat,
    sub,
    tank,
    trooper;

    public static UnitType of(String str) {
        return str != null ? valueOf(str) : null;
    }
}

File: src/main/kotlin/com/github/weewar/mapviewer/model/enums/UnitType.kt
package com.github.weewar.mapviewer.model.enums

enum class UnitType {
    antiAircraft,
    assaultArtillery,
    battleship,
    berserker,
    bomber,
    destroyer,
    dfa,
    heavyArtillery,
    heavyTank,
    heavyTrooper,
    helicopter,
    hovercraft,
    jet,
    lightArtillery,
    raider,
    speedboat,
    sub,
    tank,
    trooper;

    companion object {
        fun of(str: String?): UnitType? {
            return if (str != null) valueOf(str) else null
        }
    }
}
